server/src/main/java/org/opensearch/cluster/routing/OperationRouting.java

```diff
@@ -300,7 +334,9 @@ private ShardIterator shardRoutings(
         @Nullable ResponseCollectorService collectorService,
         @Nullable Map<String, Long> nodeCounts
     ) {
-        if (ignoreAwarenessAttributes()) {
+        if (isWeightedRoundRobinEnabled()) {
+            return indexShard.activeInitializingShardsWRR(fetchWRRWeights(), nodes, wrrShardsCache);
```

Comment:
can we get metadata only once rather than twice, maybe moving the fetch logic Metadata will have avoid two private methods doing redundant operations